core/iam_authenticator.go

```diff
@@ -533,23 +537,28 @@ func newIamTokenData(tokenResponse *IamTokenServerResponse) (*iamTokenData, erro
 }
 
 // isTokenValid: returns true iff the IamTokenData instance represents a valid (non-expired) access token.
-func (this *iamTokenData) isTokenValid() bool {
-	if this.AccessToken != "" && GetCurrentTime() < this.Expiration {
+func (td *iamTokenData) isTokenValid() bool {
+	// We'll use "exp - 10" so that we'll treat an otherwise valid (unexpired) access token
```

Comment:
This is the key to the fix... the isTokenValid() function belonging to the "iamTokenData" struct.